nck/clients/adobe_client.py

```diff
@@ -0,0 +1,91 @@
+# GNU Lesser General Public License v3.0 only
+# Copyright (C) 2020 Artefact
+# licence-information@artefact.com
+#
+# This program is free software; you can redistribute it and/or
+# modify it under the terms of the GNU Lesser General Public
+# License as published by the Free Software Foundation; either
+# version 3 of the License, or (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+# Lesser General Public License for more details.
+#
+# You should have received a copy of the GNU Lesser General Public License
+# along with this program; if not, write to the Free Software Foundation,
+# Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+
+import logging
+import datetime
+import requests
+import jwt
+
+IMS_HOST = "ims-na1.adobelogin.com"
+IMS_EXCHANGE = "https://ims-na1.adobelogin.com/ims/exchange/jwt"
+DISCOVERY_URL = "https://analytics.adobe.io/discovery/me"
+
+logging.basicConfig(level="INFO")
+logger = logging.getLogger()
+
+
+class AdobeClient:
+    """
+    Create an Adobe Client for JWT Authentification,
+    following the steps described in this repo:
+    https://github.com/AdobeDocs/analytics-2.0-apis/tree/master/examples/jwt/python
+    """
+
+    def __init__(
+        self,
+        client_id,
+        client_secret,
+        tech_account_id,
+        org_id,
+        private_key_path,
+        global_company_id,
+    ):
+        self.client_id = client_id
+        self.client_secret = client_secret
+        self.tech_account_id = tech_account_id
+        self.org_id = org_id
+        self.private_key_path = private_key_path
+        self.global_company_id = global_company_id
+
+        # Creating jwt_token attribute
+        logging.info("Getting jwt_token.")
+        with open(self.private_key_path, "r") as file:
+            private_key = file.read()
```

Comment:
Est-ce que renseigner un path vers un fichier s'imbriquera bien avec notre architecture Airflow?

Reply:
Hmm oui, il faudrait mettre le fichier dans le cluster, au bon endroit, et Berlioz n'est pas fait pour ça encore.
On peut pas avoir les credentials dans la requête comme pour les autres connecteurs?

Reply:
Jusqu'à présent quand je mettais la clé en dur dans la commande, ça plantait (ValueError: Could not deserialize key data). Je vais creuser ça, c'est peut-être une histoire d'encoding.